www/js/app.js

```diff
@@ -79,8 +83,16 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     params['useCache'] = settingsStore.getItem('useCache') !== 'false';
     // A parameter to set the app theme and, if necessary, the CSS theme for article content (defaults to 'light')
     params['appTheme'] = settingsStore.getItem('appTheme') || 'light'; // Currently implemented: light|dark|dark_invert|dark_mwInvert
-    document.getElementById('appThemeSelect').value = params.appTheme;
-    uiUtil.applyAppTheme(params.appTheme);
+     document.getElementById('appThemeSelect').value = params.appTheme;
+     if (params.appTheme =="auto_invert") {
+        handleAutoGenTheme();
+     }
+     else if (params.appTheme =="auto_mwInvert") {
+         handleAutoWikiTheme();
+      }
+     else uiUtil.applyAppTheme(params.appTheme);
+     document.getElementById('appThemeSelect').value = params.appTheme;
```

Comment:
All the handling should be done inside `uiUtil.applyAppTheme()`. Don't make a separate handling function for each possible value of `appThemeSelect` here in app.js! You clutter app.js unnecessarily, when we already have a generic handler. You shouldn't need to add any code here at the head of the file.

Reply:
Yeah, I cluttered the app by repeating the same code. So I added the code in `applyAppTheme()` .

Reply:
You should only need one line here:

`uiUtil.applyAppTheme(params.appTheme);`

and the rest you have added can go into `uiUtil.applyAppTheme()`.